libp2p/protocols/pubsub/gossipsub/scoring.nim

```diff
@@ -253,6 +253,12 @@ proc updateScores*(g: GossipSub) = # avoid async
 
   trace "updated scores", peers = g.peers.len
 
+proc scoringHeartbeat*(g: GossipSub) {.async.} =
+  while g.heartbeatRunning:
```

Comment:
`await  sleepasync` will raise on `scoringHeartbeatFut.cancel` so the boolean is redundant

Reply:
`while not scoringHeartbeatFut.finished` then